docs/src/main/tut/faq.md

```diff
@@ -0,0 +1,261 @@
+---
+layout: page
+title:  "FAQ"
+section: "faq"
+position: 3
+---
+
+# Frequently Asked Questions
+
+Feel free to add new questions and to ping [@Scalding](http://twitter.com/scalding) for an answer.
+
+# Running Scalding
+
+### Who actually uses Scalding?
+
+Twitter uses it in production all over the place!
+
+Check out our [Powered By](powered_by.html) page for more examples.
+
+### I'm having trouble with scald.rb, and I just want to run jars in my own system:
+
+See this [conversation on Twitter](https://twitter.com/Joolz/status/264834261549457409).
+
+### Can Scalding be run on Amazon's Elastic MapReduce?
+
+Yes! See the [cascading-user group discussion](https://groups.google.com/forum/?fromgroups#!topic/cascading-user/5RfJa8n1JPo).  We would like to see someone prepare a patch for scald.rb to handle submission of scalding jobs to EMR.
+
+### Scalding complains when I use a [TimePathedSource](https://github.com/twitter/scalding/blob/master/src/main/scala/com/twitter/scalding/FileSource.scala#L213) and some of the data is missing. How can I ignore that error?
+
+Pass the option `--tool.partialok` to your job and it will ignore any missing data. It's safer to work around by either filling with place-holder empty files, or writing sources thatxb will skip known-missing dates. Using that option by default is very dangerous.
+
+### I receive this error when running `sbt update`: <tt>Error occurred during initialization of VM. Incompatible minimum and maximum heap sizes specified</tt>
+
+In your sbt script, set `local min=$(( $mem / 2 ))`
+
+# Writing Jobs
+
+### How do I perform windowed calculations (for example, moving average) in Scalding?
+
+You want to use `GroupBuilder.scanLeft`.  A `scanLeft` is like a `foldLeft` except that you output each intermediate value.  Both of these functions are part of the standard Scala library as well. [See StackOverflow for `scanLeft` examples](http://stackoverflow.com/questions/6799441/in-scala-how-do-i-keep-track-of-running-totals-without-using-var).  For the specific example of moving averages in Scalding, see the [cascading-user group discussion](https://groups.google.com/forum/?fromgroups#!topic/cascading-user/CTYOjlHs6xE).
+
+### How do I read a single reduced value from a pipe?
```

Comment:
this is no longer accurate. There is Execution and ValuePipe, both are related to this problem.